index.js

```diff
@@ -1,8 +1,106 @@
 const formEncode = require('form-urlencoded');
+const promisify = require('util.promisify');
+const debug = require('debug')('token-introspection');
+const JwksClient = require('jwks-rsa');
+const jwt = require('jsonwebtoken');
+const jwk2pem = require('pem-jwk').jwk2pem;
+
+const jwtVerify = promisify(jwt.verify);
+
+async function localIntrospect(keys, allowedAlgorithms, token, tokenTypeHint) {
+  function findCandidateKeys(jwtHeader) {
```

Comment:
This function can be put into same module in root instead, accepting all as params instead of closure.